sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeCreator.scala

```diff
@@ -77,7 +77,7 @@ case class GpuCreateArray(children: Seq[Expression], useStringTypeWhenEmpty: Boo
         }
       }
       GpuColumnVector.from(ColumnVector.makeList(numRows,
-        GpuColumnVector.getNonNestedRapidsType(dataType.elementType),
+        GpuColumnVector.getArrayNestedRapidsType(dataType.elementType),
```

Comment:
```suggestion
      GpuColumnVector.from(ColumnVector.makeList(numRows,
        dataType.elementType match {
          case _:ArrayType => DType.LIST
          case _ => GpuColumnVector.getNonNestedRapidsType(dataType.elementType)
        },
        columns: _*), dataType)
```

Reply:
This is NOT going to work. We need CUDF API changes.  makeList needs the dataType so it can know what to do if we are creating an empty column (no rows).  I'm not sure how often that will happen in practice but if it does happen this code will produce a bogus column. The type will not make any since because there will be a LIST column with no child data column.

Reply:
I added new comments for what I want.  Really all it ends up being is a comment in the code explaining why the situation and why it is OK to do what we are doing.